### PR TITLE
Remove unused functions from _foundation

### PR DESCRIPTION
**Pull request by nicoddemus**
Also rename `_foundation` to `_util`, which is only a source of bad memories.

### CODE REVIEW

**Comment by tigarmo** [comment]
that was very fast

**Comment by nicoddemus** [comment]
🏃

File: src/barril/_util/types_.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    :raises ValueError:
+        if the format is invalid.
+    """
+    try:
+        pattern % arguments
+    except (TypeError, ValueError):
+        raise ValueError("%r is not a valid format string." % pattern)
+
+
+def IsNumber(v) -> bool:
```
Comment (igortg) [inline src/barril/_util/types_.py]: Should we replace this by `math.isnan` in the future?

Comment (nicoddemus) [reply src/barril/_util/types_.py]: This is not meant to check for NaNs, but to check if the given value is a "number", for example `True` and `"1"` are not numbers as far as this function is concerned.

**Pull request merged**